pagopa/io-app
# feat: [IOPID-1549] Revamp `LoadingSpinnerOverlay` and `LogoutScreen`

## Short description
This PR adds the new DS to `LoadingSpinnerOverlay` and `LogoutScreen`.

> [!tip]
> The `LoadingSpinnerOverlay` will be replace in the future with other loading/waiting patterns.

<details><summary>Demo</summary>
<p>

| LogoutScreen | LoadingSpinnerOverlay | LoadingSpinnerOverlay + Button | 
| - | - | - |
| <video src="https://github.com/pagopa/io-app/assets/16268789/468d0b27-e3c5-4e89-a971-7b0623b09506" /> | <video src="https://github.com/pagopa/io-app/assets/16268789/aa863726-683c-4611-b8b0-1ce44555767e" /> | <video src="https://github.com/pagopa/io-app/assets/16268789/ad48c5c5-b96d-4a41-8b4d-d6865c888c4d" /> |

</p>
</details> 

## List of changes proposed in this pull request
- Revamped `LoadingSpinnerOverlay`
- Revamped `LogoutScreen`
- Replaced `RefreshIndicator` with `LoadingIndicator` inside `IdpLoginScreen`
- Update snapshots

## How to test
- Make a login with SPID. You should see the `LoadingIndicator`.
- Make a logout commenting the `logout()` call in the `useEffect` inside `LogoutScreen`. You should see the revamped `LogoutScreen`.
- Navigate to ToS in profile section. You should see `LoadingSpinnerOverlay` while the `WebView` loads the ToS webpage.
- Try an EIC login. After the PIN insertion, before the EIC reading part, you should see the revamped `LoadingSpinnerOverlay` with an outline button.
